pkg/registry/client.go

```diff
@@ -387,8 +349,74 @@ func (client *registryClient) TagMetadata(manifest distribution.Manifest, opts *
 
 		return ti, nil
 	default:
-		return nil, fmt.Errorf("invalid manifest type")
+		return nil, fmt.Errorf("invalid manifest type %T", manifest)
+	}
+}
+
+// TagInfoFromReferences is a helper method to retrieve metadata for a given
+// list of references. It will return the most recent pushed manifest from the
+// list of references.
+func TagInfoFromReferences(client *registryClient, opts *options.ManifestOptions, logCtx *log.LogContext, ti *tag.TagInfo, references []distribution.Descriptor) (*tag.TagInfo, error) {
```

Comment:
Since this is merely an internal helper, I think it could be better prototyped as:

```go
func (client *registryClient) tagInfoFromReferences(...
```

to make it a method of `registryClient`  and make it non-exported.